smartsim/_core/launcher/step/step.py

```diff
@@ -67,9 +68,15 @@ def _create_unique_name(entity_name: str) -> str:
         return step_name
 
     def get_output_files(self) -> t.Tuple[str, str]:
-        """Return two paths to error and output files based on cwd"""
-        output = self.get_step_file(ending=".out")
-        error = self.get_step_file(ending=".err")
+        """Return two paths to error and output files based on metadata directory"""
+        try:
+            output_dir = self.meta["status_dir"]
+        except KeyError as exc:
+            raise KeyError("Status directory for this step has not been set.") from exc
+        if not osp.exists(output_dir):
+            pathlib.Path(output_dir).mkdir(parents=True, exist_ok=True)
```

Comment:
It seems a little odd to have this routine have two effects (1) return the paths to the output and error files and (2) create an output directory. Is there another place where these two lines make more sense.

Reply:
Definitely! I'll break it out into its own function and I could either call the new function from within `get_output_files` or I could call it within `_launch_step` right before I call `get_output_files`. Any preference?